src/components/macro/macro.component.ts

```diff
@@ -1,12 +1,132 @@
-import { Component } from '@angular/core';
+import { Macro } from '../../config-serializer/config-items/Macro';
+import { TextMacroAction } from '../../config-serializer/config-items/macro-action/TextMacroAction';
+import { UhkConfigurationService } from '../../services/uhk-configuration.service';
+import { MacroItemComponent } from './macro-item/macro-item.component';
+import { AfterViewInit, Component, OnDestroy, OnInit, QueryList, ViewChildren } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { DragulaService } from 'ng2-dragula/ng2-dragula';
+import { Subscription } from 'rxjs/Subscription';
 
 @Component({
     selector: 'macro',
     template: require('./macro.component.html'),
-    styles: [require('./macro.component.scss')]
+    styles: [require('./macro.component.scss')],
+    viewProviders: [DragulaService]
 })
-export class MacroComponent {
+export class MacroComponent implements OnInit, OnDestroy, AfterViewInit {
+    @ViewChildren(MacroItemComponent) macroItems: QueryList<MacroItemComponent>;
 
-    constructor() {
+    private macro: Macro;
+
+    private sub: Subscription;
+    private macroItemsSub: Subscription;
+    private addedNewAction: boolean = false;
+    private hasChanges: boolean = false;
+    private dragEnabled: boolean;
+
+    constructor(
+        private uhkConfigurationService: UhkConfigurationService,
+        private route: ActivatedRoute,
+        private dragulaService: DragulaService
+    ) {
+        /* tslint:disable:no-unused-variable: Used by Dragula. */
+        dragulaService.setOptions('macroActions', {
+            moves: function (el: any, container: any, handle: any) {
+                return handle.className.indexOf('action--movable') !== -1;
+            }
+        });
+        /* tslint:enable:no-unused-variable */
+    }
+
+    ngOnInit() {
+        this.sub = this.route.params.subscribe((params: any) => {
+            const id: number = Number(params.id);
+            this.macro = this.getMacro(id);
+            this.dragEnabled = true;
+            this.hasChanges = false;
+       });
+    }
+
+    ngAfterViewInit() {
+       this.macroItemsSub = this.macroItems.changes.subscribe((data: any) => {
```

Comment:
If my calculations are right, then `data`'s type is ` QueryList<MacroItemComponent>` here. Please, replace it.